### PR TITLE
set up caching for GitHub test workflow, and fix a typo on the history page

### PR DESCRIPTION
**Pull request by DavidBJaffe**

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -14,12 +14,44 @@ jobs:
...
       - name: install cargo-license
         run: cargo install cargo-license
+
+      # set up caching (duplicated verbatim below)
+
+      - name: Cache cargo registry
+        uses: actions/cache@v1
+        with:
+          path: ~/.cargo/registry
+          key: ${{runner.os}}-cargo-registry-${{hashFiles('Cargo.lock')}}
```
Comment (wyattmcdonnell) [inline .github/workflows/test.yaml:33]:
```suggestion
          key: ${{runner.os}}-stable-cargo-registry-${{hashFiles('**/Cargo.lock')}}
```

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -14,12 +14,44 @@ jobs:
...
+          restore-keys: ${{runner.os}}-cargo-registry-
```
Comment (wyattmcdonnell) [inline .github/workflows/test.yaml:34]:
```suggestion
          restore-keys: ${{runner.os}}-stable-cargo-registry-
```

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -14,12 +14,44 @@ jobs:
...
+      - name: Cache cargo git index
+        uses: actions/cache@v1
+        with:
+          path: ~/.cargo/git
+          key: ${{runner.os}}-cargo-git-${{hashFiles('Cargo.lock')}}
```
Comment (wyattmcdonnell) [inline .github/workflows/test.yaml:39]:
```suggestion
          key: ${{runner.os}}-stable-cargo-index-${{hashFiles('**/Cargo.lock')}}
```

File: .github/workflows/test.yaml [hunk 4]
```diff
@@ -14,12 +14,44 @@ jobs:
...
+          restore-keys: ${{runner.os}}-cargo-git-
```
Comment (wyattmcdonnell) [inline .github/workflows/test.yaml:40]:
```suggestion
          restore-keys: ${{runner.os}}-stable-cargo-index-
```

File: .github/workflows/test.yaml [hunk 5]
```diff
@@ -14,12 +14,44 @@ jobs:
...
+      - name: Cache cargo build
+        uses: actions/cache@v1
+        with:
+          path: lib/rust/target
+          key: ${{runner.os}}-cargo-target-
```
Comment (wyattmcdonnell) [inline .github/workflows/test.yaml:45]:
```suggestion
          key: ${{runner.os}}-stable-cargo-release-target-
```

File: .github/workflows/test.yaml [hunk 6]
```diff
@@ -14,12 +14,44 @@ jobs:
...
+            ${{hashFiles('Cargo.lock')}}-
+            ${{hashFiles('Cargo.toml')}}-
+            ${{hashFiles('**/*.rs')}}
+          restore-keys: |
+            ${{runner.os}}-cargo-target-${{hashFiles('Cargo.lock')}}-
+            ${{runner.os}}-cargo-target-
```
Comment (wyattmcdonnell) [inline .github/workflows/test.yaml:51]:
```suggestion
            ${{runner.os}}-stable-cargo-release-target-${{hashFiles('**/Cargo.lock')}}-
             ${{runner.os}}-stable-cargo-release-target-
```

File: .github/workflows/test.yaml [hunk 7]
```diff
@@ -14,12 +14,44 @@ jobs:
...
+          path: ~/.cargo/git
+          key: ${{runner.os}}-cargo-git-${{hashFiles('Cargo.lock')}}
+          restore-keys: ${{runner.os}}-cargo-git-
+      - name: Cache cargo build
+        uses: actions/cache@v1
+        with:
+          path: lib/rust/target
+          key: ${{runner.os}}-cargo-target-
+            ${{hashFiles('Cargo.lock')}}-
+            ${{hashFiles('Cargo.toml')}}-
```
Comment (wyattmcdonnell) [inline .github/workflows/test.yaml:47]:
```suggestion
            ${{hashFiles('**/Cargo.lock')}}-
              ${{hashFiles('**/Cargo.toml')}}-
```

File: .github/workflows/test.yaml (hunk 1 above)
Comment (sjackman) [reply .github/workflows/test.yaml:33]: There's multiple `Cargo.toml` files, but only one `Cargo.lock` file, so `**/` isn't needed here.

File: .github/workflows/test.yaml (hunk 3 above)
Comment (sjackman) [reply .github/workflows/test.yaml:39]: Ditto `Cargo.lock`

File: .github/workflows/test.yaml (hunk 7 above)
Comment (sjackman) [reply .github/workflows/test.yaml:47]: Ditto `Cargo.lock`

File: .github/workflows/test.yaml (hunk 6 above)
Comment (sjackman) [reply .github/workflows/test.yaml:51]: Ditto `Cargo.lock`

File: .github/workflows/test.yaml [hunk 8]
```diff
@@ -31,16 +63,54 @@ jobs:
...
```
Comment (sjackman) [inline .github/workflows/test.yaml:106]:
```suggestion
            ${{hashFiles('**/Cargo.toml')}}-
```

File: .github/workflows/test.yaml [hunk 9]
```diff
@@ -14,12 +14,44 @@ jobs:
...
+            ${{hashFiles('**/*.rs')}}
```
Comment (sjackman) [inline .github/workflows/test.yaml:48]: Does breaking `key` across multiple lines like this work? It may. I don't know.

Comment (DavidBJaffe) [reply .github/workflows/test.yaml:48]: I will unbreak it!

File: .github/workflows/test.yaml [hunk 10]
```diff
@@ -31,16 +63,54 @@ jobs:
       - name: unit tests
         run: cd enclone_main; cargo test --release --features basic -- --nocapture
 
+      # clean up (duplicated verbatim below)
+
+      - name: Remove the Cargo target directory
+        if: github.ref == 'refs/heads/master'
+        run: cd lib/rust && cargo clean
```
Comment (sjackman) [inline .github/workflows/test.yaml:70]: `cargo clean` needs to occur before `cargo build`.

**Pull request merged**